components/bootloader_support/src/bootloader_console.c

```diff
@@ -109,12 +118,34 @@ void bootloader_console_init(void)
     esp_rom_uart_set_as_console(ESP_ROM_UART_USB);
     esp_rom_install_channel_putc(1, bootloader_console_write_char_usb);
 }
-#endif //CONFIG_ESP_CONSOLE_USB_CDC
+#endif // CONFIG_ESP_CONSOLE_IS_USB_CDC_ENABLED
 
-#ifdef CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG
-void bootloader_console_init(void)
+//
+// Usb Serial/JTAG
+//
+#if CONFIG_ESP_CONSOLE_IS_USB_SERIAL_JTAG_ENABLED
+void bootloader_console_init_usb_serial_jtag(void)
 {
     UartDevice *uart = GetUartDevice();
     uart->buff_uart_no = ESP_ROM_USB_SERIAL_DEVICE_NUM;
 }
-#endif //CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG
+#endif // CONFIG_ESP_CONSOLE_IS_USB_SERIAL_JTAG_ENABLED
+
+//
+// Init All
+//
+void bootloader_console_init(void)
+{
+#if CONFIG_ESP_CONSOLE_IS_NONE_ENABLED
+    bootloader_console_init_none();
+#endif
+#if CONFIG_ESP_CONSOLE_IS_UART_ENABLED
+    bootloader_console_init_uart();
+#endif
+#if CONFIG_ESP_CONSOLE_IS_USB_CDC_ENABLED
+    bootloader_console_init_usb_cdc();
+#endif
+#if CONFIG_ESP_CONSOLE_IS_USB_SERIAL_JTAG_ENABLED
+    bootloader_console_init_usb_serial_jtag();
+#endif
+}
```

Comment:
Looks like here we may end up calling esp_rom_install_channel_putc (for redirecting ets_printf/esp_rom_printf) multiple times, and it will get redirected to the most recent destination. Is that the expected behavior?

Reply:
Which 2 functions both call `esp_rom_install_channel_putc()`? 

To be honest, I'm not familiar with what "redirecting ets_printf" means. 

Regardless, I only see that`bootloader_console_init_none()` and `bootloader_console_init_usb_cdc() `call `esp_rom_install_channel_putc()`, but `NONE` is only true when the others are NOT enabled, so it is not a problem